### ISSUE TITLE
Why is Task's JoinHandle !Send?

### ISSUE THREAD

**Issue by qiujiangkun** [DataDog/glommio#360]
I would like to join the JoinHandle from another thread, but the JoinHandle is !Send due to NonNull.
Is it actually Send or !Send by design?
https://github.com/najamelan/async_executors/pull/5#discussion_r641702676

**Comment by qiujiangkun** [comment]
Task's JoinHandle
```rust
error[E0277]: `NonNull<()>` cannot be sent between threads safely
   --> async_executors/src/core/join_handle.rs:262:2
    |
262 |     static_assertions::assert_impl_all!(JoinHandle<()>: Send);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |     |
    |     `NonNull<()>` cannot be sent between threads safely
    |     required by this bound in `assert_impl_all`
    |
    = help: within `join_handle::JoinHandle<()>`, the trait `std::marker::Send` is not implemented for `NonNull<()>`
    = note: required because it appears within the type `glommio::task::JoinHandle<()>`
    = note: required because it appears within the type `Option<glommio::task::JoinHandle<()>>`
    = note: required because it appears within the type `glommio::executor::multitask::Task<()>`
    = note: required because it appears within the type `glommio::Task<()>`
    = note: required because it appears within the type `Option<glommio::Task<()>>`
```

**Comment by glommer** [comment]
I left a comment in the project you linked to.
tl;dr: ufortunately a `Send` task's join handle (or anything task related really) is a big no no and would break the thread-per-core model we follow.
